internal/directives/config.go

```diff
@@ -23,6 +23,11 @@ func (c Config) DeepCopy() Config {
 	return runtime.DeepCopyJSON(c)
 }
 
+// ToJSON marshals the configuration to JSON.
+func (c Config) ToJSON() ([]byte, error) {
+	return json.Marshal(c)
+}
```

Comment:
I'm curious what the rationale for this was. I'm sure you had a reason that you wanted to call this somewhere instead of calling `json.Marshal()` directly?

Reply:
I didn't want the reconciler to end up with too much raw logic when working with the further directives abstractions, and I want to add a check that ensures an empty map actually returns nil instead of an empty JSON object.